Dockerfile.windows

```diff
@@ -0,0 +1,77 @@
+# escape=`
+
+# BUILD
+ARG BASE_IMAGE=mcr.microsoft.com/windows/servercore:ltsc2019
+FROM $BASE_IMAGE as builder
+
+# Restore the default Windows shell for correct batch processing.
+SHELL ["cmd", "/S", "/C"]
+
+# Download the Visual Studio Build Tools bootstrapper.
+ADD https://aka.ms/vs/16/release/vs_buildtools.exe C:\Temp\vs_buildtools.exe
+
+# Use the latest release channel.
+ADD https://aka.ms/vs/16/release/channel C:\Temp\VisualStudio.chman
+
+# For help on command-line syntax:
+# https://docs.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio
+# Install MSVC C++ compiler and CMake.
+RUN C:\Temp\vs_buildtools.exe `
+    --quiet --wait --norestart --nocache `
+    --installPath C:\BuildTools `
+    --channelUri C:\Temp\VisualStudio.chman `
+    --installChannelUri C:\Temp\VisualStudio.chman `
+    --add Microsoft.VisualStudio.Workload.VCTools;includeRecommended `
+    || IF "%ERRORLEVEL%"=="3010" EXIT 0
+
+RUN powershell -command "iex (new-object net.webclient).downloadstring('https://get.scoop.sh'); `
+  scoop install winflexbison"
+
+ENV configuration=Release msvc="Visual Studio 16 2019"
+RUN setx /M PATH "%PATH%;C:\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin"
+
+COPY . /src
+
+WORKDIR /src/build
+
+# CACHE GENERATION
+RUN powershell -Command cmake -G "$ENV:msvc" -DCMAKE_BUILD_TYPE="$ENV:configuration" `
+                     -DCIO_BACKEND_FILESYSTEM=Off `
+                     -DFLB_TESTS_INTERNAL=On `
+                     -D FLB_WITHOUT_flb-rt-out_elasticsearch=On `
+                     -D FLB_WITHOUT_flb-rt-out_td=On `
+                     -D FLB_WITHOUT_flb-rt-out_forward=On `
+                     -D FLB_WITHOUT_flb-rt-in_disk=On `
+                     -D FLB_WITHOUT_flb-rt-in_proc=On `
+                     -D FLB_WITHOUT_flb-it-parser=On `
+                     -D FLB_WITHOUT_flb-it-unit_sizes=On `
+                     -D FLB_WITHOUT_flb-it-network=On `
+                     -D FLB_WITHOUT_flb-it-pack=On `
+                     ../
```

Comment:
I think we don't need to build tests on Docker (`FLB_TESTS_INTERNAL=On`).
This means that you can simplify this build command to:

```
RUN powershell -Command cmake -G "$ENV:msvc" -DCMAKE_BUILD_TYPE="$ENV:configuration" ..
```